jax/lax_linalg.py

```diff
@@ -644,6 +645,63 @@ def lu_pivots_to_permutation(swaps, m):
   return result
 
 
+def _lu_solve_core(lu, pivots, b, trans):
+  m = lu.shape[0]
+  permutation = lu_pivots_to_permutation(pivots, m)
+  x = np.reshape(b, (m, -1))
+  if trans == 0:
+    x = x[permutation, :]
+    x = triangular_solve(lu, x, left_side=True, lower=True, unit_diagonal=True)
+    x = triangular_solve(lu, x, left_side=True, lower=False)
+  elif trans == 1 or trans == 2:
+    conj = trans == 2
+    x = triangular_solve(lu, x, left_side=True, lower=False, transpose_a=True,
+                         conjugate_a=conj)
+    x = triangular_solve(lu, x, left_side=True, lower=True, unit_diagonal=True,
+                         transpose_a=True, conjugate_a=conj)
+    x = x[np.argsort(permutation), :]
+  else:
+    raise ValueError("'trans' value must be 0, 1, or 2, got {}".format(trans))
+  return lax.reshape(x, b.shape)
+
+
+@partial(api.jit, static_argnums=(3,))
+def _lu_solve(lu, pivots, b, trans):
+  from .experimental.vectorize import vectorize  # avoid recursive import
+  if len(lu.shape) < 2 or lu.shape[-1] != lu.shape[-2]:
+    raise ValueError("last two dimensions of LU decomposition must be equal, "
+                     "got shape {}".format(lu.shape))
+  if len(b.shape) < 1:
+    raise ValueError("b matrix must have rank >= 1, got shape {}"
+                     .format(b.shape))
+  # Broadcasting follows NumPy's convention for linalg.solve: the RHS is
+  # treated as a (batched) vector if the number of dimensions differ by 1.
+  # Otherwise, broadcasting rules apply.
+  rhs_vector = lu.ndim == b.ndim + 1
+  if rhs_vector:
+    if b.shape[-1] != lu.shape[-1]:
+      raise ValueError("When LU decomposition matrix and b have the same "
+                       "number of dimensions, last axis of LU decomposition "
+                       "matrix (shape {}) and b array (shape {}) must match"
+                       .format(lu.shape, b.shape))
+    b = b[..., np.newaxis]
+  else:
+    if b.shape[-2] != lu.shape[-1]:
+      raise ValueError("When LU decomposition matrix and b different "
+                       "numbers of dimensions, last axis of LU decomposition "
+                       "matrix (shape {}) and second to last axis of b array "
+                       "(shape {}) must match"
+                       .format(lu.shape, b.shape))
+  solve_once = partial(_lu_solve_core, trans=trans)
+  x = vectorize('(n,n),(n),(n,k)->(n,k)')(solve_once)(lu, pivots, b)
```

Comment:
Another option would be to use `vmap` here?

Although I'm guessing that since you also sent a PR adding `jax.numpy.vectorize` that's an option too.

Reply:
The downside of using `vmap` is that then I need to add figured out the broadcast shape and add explicit broadcasting for all the input arguments, both of which `vectorize` does for me.

I'd also be happy to wait until after the `vectorize` PR is finished here, though...

Reply:
I rewrote to use `jax.numpy.vectorize` instead, which avoids the need for the local import.